flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierUnaligner.java

```diff
@@ -101,10 +101,8 @@ public void processBarrier(CheckpointBarrier barrier, InputChannelInfo channelIn
 			allBarriersReceivedFuture = new CompletableFuture<>();
 			checkpointCoordinator.initCheckpoint(barrierId, barrier.getCheckpointOptions());
 
-			for (final InputGate gate : inputGates) {
-				for (int index = 0, numChannels = gate.getNumberOfInputChannels(); index < numChannels; index++) {
-					gate.getChannel(index).checkpointStarted(barrier);
-				}
+			for (final BlockableInput input : inputs) {
+				input.checkpointStarted(barrier);
```

Comment:
Why do we block source inputs when we receive a barrier from a non-source input?
(maybe a comment is missing here or for `StreamTaskSourceInput.checkpointStarted`)

Reply:
Good question. I've added a larger explanation why in `StreamTaskSourceInput#checkpointStarted`

Reply:
Thanks!